src/v/kafka/server/usage_aggregator.cc

```diff
@@ -109,16 +109,34 @@ usage& usage::operator+=(const usage& other) {
     return *this;
 }
 
-template<typename time_point>
-auto epoch_time_secs(time_point now) {
+/// The type that will be providing timestamps in form of seconds since
+/// epoch. ss::lowres_clock has no concept of epoch time, therefore it isn't
+/// used for generating timestampsp
+template<typename clock_type>
+using ts_type = std::conditional_t<
+  std::is_same_v<clock_type, ss::lowres_clock>,
+  ss::lowres_system_clock,
+  clock_type>;
```

Comment:
what is being protected here? it seems like you could add a static assertion somewhere that a clock type is not a lowres clock. this particular formulation seems to allow users to plug in a lowres clock (mistake?) but then things compile fine and a different clock is transparently swapped in?